corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -20,30 +20,47 @@
 PagedResult = namedtuple('PagedResult', 'total hits')
 
 
-def get_last_submission_time_for_user(domain, user_id, datespan):
-    form_query = FormES() \
-        .domain(domain) \
-        .user_id([user_id]) \
-        .completed(gte=datespan.startdate.date(), lte=datespan.enddate.date()) \
-        .sort("form.meta.timeEnd", desc=True) \
-        .size(1)
-    results = form_query.run().hits
-
+def get_last_submission_time_for_user(domain, user_ids, datespan):
     def convert_to_date(date):
         return string_to_datetime(date).date() if date else None
 
-    return convert_to_date(results[0]['form']['meta']['timeEnd'] if results else None)
+    query = (
```

Comment:
This is mostly a simplified version of https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/analytics/esaccessors.py#L131-L176 that only returns `{user_id: timeEnd}`